### ISSUE TITLE
Focus ordering wrong when docking to bottom

### ISSUE THREAD

**Issue by darrenburns** [Textualize/textual#502]
If you have 2 widgets docked to the bottom of the screen, cycle focus forward, the focus moves from bottom to top of the screen instead of top to bottom.

**Comment by willmcgugan** [comment]
That makes sense. They are that order in the DOM.

I think we will need some way of explicitly defining the tab order.

**Comment by willmcgugan** [comment]
Keep it open. We will need to figure out the best way to tackle this.

**Comment by willmcgugan** [comment]
I'm thinking that the tab order should be pre-calculated by collecting the focusable things and sorting left-to-right/top-to-bottom.

This should make the tab order correct, no matter the layout.

We should also allow for right-to-left ordering for future rtol language support.
